android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java

```diff
@@ -413,22 +410,34 @@ public void getAlbums(final ReadableMap params, final Promise promise) {
         WritableArray response = new WritableNativeArray();
         try {
           if (media.moveToFirst()) {
+            Map<String, Integer> albums = new TreeMap<>();
             do {
-              String albumName = media.getString(media.getColumnIndex(bucketDisplayName));
-              int count = media.getInt(media.getColumnIndex("count"));
+              String albumName = media.getString(media.getColumnIndex(MediaStore.Images.ImageColumns.BUCKET_DISPLAY_NAME));
+              if (albumName != null) {
+                Integer albumCount = albums.get(albumName);
+                if (albumCount == null) {
+                  albums.put(albumName, 1);
+                } else {
+                  albums.put(albumName, albumCount + 1);
+                }
+              }
+            } while (media.moveToNext());
+
+            for (Map.Entry<String, Integer> albumEntry : albums.entrySet()) {
               WritableMap album = new WritableNativeMap();
-              album.putString("title", albumName);
-              album.putInt("count", count);
+              album.putString("title", albumEntry.getKey());
+              album.putInt("count", albumEntry.getValue());
               response.pushMap(album);
-            } while (media.moveToNext());
+            }
           }
         } finally {
           media.close();
           promise.resolve(response);
         }
       }
-    } catch (Exception e) {}
-
+    } catch (Exception e) {
```

Comment:
Could you consider to merge 2 try catch block into 1? the original implementation got problem here.

Reply:
the best would be using the try-with-resources construct for the Cursor but is supported from API Level 19, If you do one main try you would need to have the `media` cursor reference outside of the try and check the nullability on the finally. I think I would rather keep as it is.